integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/scaletest/TestReport.scala

```diff
@@ -34,14 +33,21 @@ class TestReport(config: Config, executionElapseMap: Map[String, Seq[Long]]) {
           s"--overwrite argument to force overwrite.")
       }
     }
-    val writer = new BufferedWriter(new FileWriter(config.reportPath))
-    writer.write("query,iteration_elapses/millis,average_elapse/millis")
-    writer.newLine()
-    executionElapseMap.foreach { case (key, value) =>
-      writer.write(s"$key,[${value.mkString(",")}], ${value.sum/(value.length)}")
-      writer.newLine()
+    val data = executionElapseMap.map { case (key, value) =>
+      (key, value, value.sum/value.length)
+    }.toSeq
+    import spark.implicits._
```

Comment:
I am not happy with reusing spark to write the data out as JSON. I know it is simple because we have access to Spark, so I am fine with it for the short term. But at least please file a follow on issue to switch this over to serializing the JSON ourselves possibly using the same library as Spark, because it is on the classpath.

Reply:
This is not difficult, I just finished this in this PR. Updated.